include/nlohmann/json.hpp

```diff
@@ -4187,27 +4216,55 @@ class basic_json // NOLINT(cppcoreguidelines-special-member-functions,hicpp-spec
         return ptr.get_unchecked(this);
     }
 
+    template<typename BasicJsonType>
+    JSON_HEDLEY_DEPRECATED(3.10.6)
+    reference operator[](const ::nlohmann::json_pointer<BasicJsonType>& ptr)
+    {
+        return ptr.get_unchecked(this);
```

Comment:
I need to double check the friend declarations for this case. I remember removing one after postponing these backwards compatibility changes.
Also make sure the unit tests really cover this properly.